app/models/challenge.rb

```diff
@@ -1,6 +1,8 @@
 class Challenge < ApplicationRecord
   belongs_to :owner, required: true, class_name: "User"
 
+  has_many :models, dependent: :nullify
```

Comment:
This may not be a good idea. When the challenge is removed everything should be cleaned:

```suggestion
  has_many :models, dependent: :destroy
```